### PR TITLE
Colored Kinect Fusion

### PR DESCRIPTION
**Pull request by DumDereDum**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/rgbd/src/tsdf_functions.cpp [hunk 1]
```diff
@@ -418,6 +418,309 @@ void integrateVolumeUnit(
...
+                    v_float32x4 projectedRGB = v_muladd(camPixVec, rgb_vfxy, rgb_vcxy);
+                    // leave only first 2 lanes
+                    projectedRGB = v_reinterpret_as_f32(v_reinterpret_as_u32(projected) &
+                        v_uint32x4(0xFFFFFFFF, 0xFFFFFFFF, 0, 0));
+
+                    // norm(camPixVec) produces double which is too slow
+                    int _u = (int)projected.get0();
+                    int _v = (int)v_rotate_right<1>(projected).get0();
+                    int rgb_u = (int)projected.get0();
+                    int rgb_v = (int)v_rotate_right<1>(projected).get0();
```
Comment (savuor) [inline modules/rgbd/src/tsdf_functions.cpp:571]: Vectorized code diverges with scalar one in what coordinate to use for RGB

File: modules/rgbd/src/tsdf_functions.cpp [hunk 2]
```diff
@@ -418,6 +418,309 @@ void integrateVolumeUnit(
...
+
+                        RGBTsdfVoxel& voxel = volDataY[z * volStrides[2]];
+                        WeightType& weight = voxel.weight;
+                        TsdfType& value = voxel.tsdf;
+                        ColorType& r = voxel.r;
+                        ColorType& g = voxel.g;
+                        ColorType& b = voxel.b;
+
+                        // update RGB
+                        if (abs(((float)(r + g + b)) - (colorRGB[0] + colorRGB[1] + colorRGB[2])) < 1000 || weight < 1)
```
Comment (savuor) [inline modules/rgbd/src/tsdf_functions.cpp:595]: What's the purpose of this extra comparison and why 1000?

Comment (DumDereDum) [reply modules/rgbd/src/tsdf_functions.cpp:595]: it is the previous version of code
will be removed

**Comment by savuor** [comment]
While there are things to do (speed up using OpenCL, rewrite raycasting to get better color filtering, fix demo to allow unregistered depth-rgb pairs) they have more sense to be implemented in upcoming PRs than in this one.
Here we have demo working and tests passing which is going to be enough for now.

**Comment by savuor** [comment]
👍

**Pull request merged**